server/src/server.ts

```diff
@@ -1,63 +1,89 @@
 import 'reflect-metadata';
-import Koa from 'koa';
 import bodyParser from 'koa-bodyparser';
+import { koaMiddleware } from '@as-integrations/koa';
 import patchKoaQs from 'koa-qs';
-import config from './common/setup/config';
+import jwt from 'koa-jwt';
+
+import config, { readSecret } from './common/setup/config';
 import getDb from './common/setup/db';
 import log from './common/setup/log';
-import dbTransactionMiddleware from './infra/middlewares/dbTransactionMiddleware';
-import errorHandlerMiddleware from './infra/middlewares/errorHandlerMiddleware';
-import initSentry from './infra/middlewares/initSentry';
-import requestLoggerMiddleware from './infra/middlewares/requestLoggerMiddleware';
 import restMiddlewares from './rest';
 import apollo from './graphql';
 import { initFunctionHandler } from './function';
+import { httpServer, server } from './http';
+import initSentry from './infra/middlewares/initSentry';
+import requestLoggerMiddleware from './infra/middlewares/requestLoggerMiddleware';
+import errorHandlerMiddleware from './infra/middlewares/errorHandlerMiddleware';
+import dbTransactionMiddleware from './infra/middlewares/dbTransactionMiddleware';
 
 // Sentry
 initSentry();
 
-// Koa
-const server = new Koa();
-patchKoaQs(server); // Adds support for query parameter nesting
-
-// Request state prototype
-server.use(async (ctx, next) => {
-  ctx.state = ctx.state || {};
-  ctx.state.log = log;
-  ctx.state._db = await getDb();
-  await next();
-});
+// Start the server or function handler
+let handler = null;
 
-// Middlewares
-server.use(requestLoggerMiddleware); // Assume no errors in logging
-server.use(errorHandlerMiddleware);
-server.use(bodyParser()); // Required by dbTransactionMiddleware.isGraphQLQuery
-server.use(dbTransactionMiddleware);
+async function setupServer() {
+  patchKoaQs(server); // Adds support for query parameter nesting
 
-// REST API routing
-restMiddlewares.forEach((middleware) => {
-  server.use(middleware);
-});
+  // Request state prototype
+  server.use(async (ctx, next) => {
+    ctx.state = ctx.state || {};
+    ctx.state.log = log;
+    ctx.state._db = await getDb();
+    ctx.state.locale =
+      ctx.acceptsLanguages('fi', 'en') || config.DEFAULT_LOCALE;
```

Comment:
Could these `'fi', 'en'` come from some constants/config file so it's easier to change them?